operator/helper/multiline.go

```diff
@@ -226,16 +212,53 @@ func SplitNone(maxLogSize int) bufio.SplitFunc {
 	}
 }
 
-// NewLineEndSplitFunc creates a bufio.SplitFunc that splits an incoming stream into
-// tokens that end with a match to the regex pattern provided
-func NewLineEndSplitFunc(re *regexp.Regexp, flushAtEOF bool, force *Flusher) bufio.SplitFunc {
+func flusherSplitFunc(force *Flusher, splitFunc bufio.SplitFunc) bufio.SplitFunc {
 	return func(data []byte, atEOF bool) (advance int, token []byte, err error) {
-		if force != nil && force.ShouldFlush() {
-			force.Flush()
-			token = trimWhitespaces(data)
-			advance = len(data)
+		advance, token, err = splitFunc(data, atEOF)
+
+		// Return as it is in case of error
+		if err != nil {
 			return
 		}
+
+		// Do not return empty log
+		if len(token) == 0 {
+			token = nil
+		}
+
+		// Return token
+		if token != nil {
+			if force != nil {
+				// Inform flusher that we just flushed
+				force.Flushed()
+			}
+			return
+		}
+
+		// If there is no token, force flush eventually
+		if force != nil {
```

Comment:
To keep things flatter, we can just return here when necessary.

```suggestion
		if force == nil {
			return
		}	
```